src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -771,6 +775,34 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 					// to get the behaviour people expect.
 					AiNodeSetArray( node, "trace_sets", AiArray( 1, 1, AI_TYPE_STRING, "__none__" ) );
 				}
+
+				if( m_linkedLights )
+				{
+					const std::vector<std::string> &lightNames = m_linkedLights->readable();
+
+					std::vector<AtNode*> lightNodesVector;
+					for ( IECore::StringVectorData::ValueType::const_iterator it = lightNames.begin(); it != lightNames.end(); ++it )
+					{
+						std::string lightName = "light:" + *(it);
+						AtNode *lightNode = AiNodeLookUpByName( lightName.c_str() );
+						if( lightNode )
+						{
+							lightNodesVector.push_back( lightNode );
+						}
+					}
+
+					AtArray *linkedLightNodes = AiArrayAllocate( lightNodesVector.size(), 1, AI_TYPE_NODE );
+					for( std::vector<AtNode*>::size_type i = 0; i < lightNodesVector.size(); ++i )
+					{
+						AiArraySetPtr( linkedLightNodes, i, lightNodesVector[i] );
+					}
```

Comment:
Avoiding NULL pointers here might be unnecessary because we intersect with the light set in `EvaluateLightLinks` now. Do you still want to keep this?